pkg/controllers/repositoryserver/repositoryserver_controller_test.go

```diff
@@ -134,24 +180,278 @@ func (s *RepoServerControllerSuite) CreateKopiaTLSSecret(data map[string][]byte)
+	}{
+		{
+			description: "Invalid Repository Password",
+			testFunction: func(rs *v1alpha1.RepositoryServer) {
+				InvalidRepositoryPassword, err := s.CreateRepositoryPasswordSecret(testutil.GetRepoPasswordSecretData("invalidPassword"))
+				c.Assert(err, IsNil)
+
+				rs.Spec.Repository.PasswordSecretRef.Name = InvalidRepositoryPassword.Name
+				rs.Spec.Repository.PasswordSecretRef.Namespace = InvalidRepositoryPassword.Namespace
+			},
+		},
+		{
+			description: "Invalid Storage Location",
+			testFunction: func(rs *v1alpha1.RepositoryServer) {
+				storageLocationData := testutil.GetDefaultS3CompliantStorageLocation()
+				storageLocationData[repositoryserver.BucketKey] = []byte("invalidbucket")
+
+				InvalidStorageLocationSecret, err := s.CreateStorageLocationSecret(storageLocationData)
+				c.Assert(err, IsNil)
+
+				rs.Spec.Storage.SecretRef.Name = InvalidStorageLocationSecret.Name
+				rs.Spec.Storage.SecretRef.Namespace = InvalidStorageLocationSecret.Namespace
+			},
+		},
+		{
+			description: "Invalid Storage location credentials",
+			testFunction: func(rs *v1alpha1.RepositoryServer) {
+				storageLocationCredsData := testutil.GetDefaultS3StorageCreds()
+				storageLocationCredsData[secrets.AWSAccessKeyID] = []byte("testaccesskey")
+
+				InvalidStorageLocationCrdesSecret, err := s.CreateStorageLocationSecret(storageLocationCredsData)
+				c.Assert(err, IsNil)
+
+				rs.Spec.Storage.CredentialSecretRef.Name = InvalidStorageLocationCrdesSecret.Name
+				rs.Spec.Storage.CredentialSecretRef.Namespace = InvalidStorageLocationCrdesSecret.Namespace
+			},
+		},
+	} {
+		invalidCR := *originalrepoServerCR
+		tc.testFunction(&invalidCR)
+
+		repoServerCRCreated, err := s.crCli.RepositoryServers(s.repoServerControllerNamespace).Create(ctx, &invalidCR, metav1.CreateOptions{})
+		c.Assert(err, IsNil)
+
+		err = s.waitOnRepositoryServerState(c, repoServerCRCreated.Name)
+		c.Assert(err, NotNil)
+
+		//Get repository server CR with the updated server information
+		repoServerCRCreated, err = s.crCli.RepositoryServers(s.repoServerControllerNamespace).Get(ctx, repoServerCRCreated.Name, metav1.GetOptions{})
+		c.Assert(err, IsNil)
+
+		c.Assert(repoServerCRCreated.Status.Progress, Equals, v1alpha1.Failed)
+	}
+}
+
+func (s *RepoServerControllerSuite) waitForRepoServerInfoUpdateInCR(repoServerName string) error {
+	ctxTimeout := 25 * time.Minute
+	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+	defer cancel()
+	err := poll.Wait(ctx, func(ctx context.Context) (bool, error) {
+		repoServerCR, err := s.crCli.RepositoryServers(s.repoServerControllerNamespace).Get(ctx, repoServerName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if repoServerCR.Status.ServerInfo.PodName == "" || repoServerCR.Status.ServerInfo.ServiceName == "" {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return errors.Wrapf(err, "repository server CR not set")
```

Comment:
The correct message here would be `podname and service name are not set` or `repo server CR not set`?